### PR TITLE
Scatter

### LINKED ISSUE #97: Add option to plot heat map of circles instead of rectangles using geom_point
Looking at the source code, it seems to me a pretty simple addition to the add the option of creating the heat map using geom_point instead of geom_tile. Check out the link see a demonstration of what I mean. http://www.nature.com/nri/journal/v14/n4/fig_tab/nri3642_F3.html

### LINKED ISSUE #124: draw_cellnote=T prints "Aa trace 2" and dots
### when `hide_colorbar = T` but only "Aa trace 2" when `hide_colorbar = F`

    D<-c("Pla", "Ond","Gra", "Dol","Tro", "Ond+Dex", "Pal","Ram", "Ond+Drop",  "Ond+Met", "Gra+Dex",  "Pal+Dex", "Dol+Dex", "Dol+Drop", "Gran+Drop")
    S1<- c(42.8, 54.6, 40.7, 81.5, 54.6, 83.3, NA, 35.3, NA, NA, 2.9, NA, NA, 54.2, NA)
    S2<-c(51.9, 64.9, 93.5, 27.7, 35.3, NA, NA, NA, NA, NA, NA, NA, 26.6, NA, NA)
    S3<-c(0.8, 25.4, 44.8, 13.3, 23.2, 71.9, 54.9, 51.3, 65.4, 52.8, 81.2, 43.7, 72.8, 76.8, 71.7)
    S4<-c(4.6, 29.4, 54.3, 22, 42.2, 76, 30.1, 48.4, 84.6, 78.8, 84.3, 7.8, 73.9, 68.8, 44.8)
    S5<-c(0.1, 20.1, 42.5, 37.7, 16.3, 63, 72.3, 34.9, 76.9, NA, 86.3, 67, NA, 71.9, 61.1)
    mydata<-data.frame(D, S4, S3, S5, S2, S1)
    rownames(mydata) <- mydata[,1]
    mydata <- mydata[,-1]
    library(heatmaply)
    heatmaply(mydata, dendrogram = F, draw_cellnote = T, hide_colorbar = T)

### LINKED ISSUE #131: Personalized hoverinfo
It seems that hoverinfo field cannot be personalized with any kind of text, looking at l162-178 in plots.R file.

Do you think it could be possible to add heatmaply parameter to specify as a character matrix personalized displayed text like cellnote field?

Thks
++
J.

### PR DESCRIPTION
**Pull request by Alanocallaghan**
Superset of #127, #98
- Close #131 
- close #97 
- close #124

### CODE REVIEW

File: R/heatmapr.R [hunk 1]
```diff
@@ -96,8 +96,8 @@
 #' "OLO" (Optimal leaf ordering, optimizes the Hamiltonian path length that is restricted by the dendrogram structure - works in O(n^4) )
 #' "mean" (sorts the matrix based on the reorderfun using marginal means of the matrix. This is the default used by \link[gplots]{heatmap.2}),
 #' "none" (the default order produced by the dendrogram),
-#' "GW" (Gruvaeus and Wainer heuristic to optimize the Hamiltonian path length that is restricted by the dendrogram structure)
-#'
+#' "GW" (Gruvaeus and Wainer heuristic to optimze the Hamiltonian path length that is restricted by the dendrogram structure)
```
Comment (talgalili) [inline R/heatmapr.R]: typo

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,5 @@
+<<<<<<< HEAD
```
Comment (talgalili) [inline NEWS.md]: No need to fix this since I create NEWS.md automatically from NEWS.

File: NEWS [hunk 3]
```diff
@@ -32,21 +48,15 @@ heatmaply 0.13.0 (2017-11-11)
...
-* heatmaply
-  - Add `cellnote_size` argument, controlling the font size of the cellnote.
-
 ### BUG FIXES
 * heatmaply
   - removing tick labels with `showticklabels` now removes the ticks as
     well as the tick labels.
+
+
+=======
```
Comment (talgalili) [inline NEWS]: Should be removed.

File: NEWS [hunk 4]
```diff
@@ -32,21 +48,15 @@ heatmaply 0.13.0 (2017-11-11)
...
 
 
-
-
-
 heatmaply 0.12.1 (2017-10-25)
 ================
-
-
-### NEW FEATURES
```
Comment (talgalili) [inline NEWS]: This should probably be kept here.

File: R/heatmaply.R [hunk 5]
```diff
@@ -563,7 +577,21 @@ heatmaply.default <- function(x,
   }
 
   # this is to fix the error: "argument * matches multiple formal arguments"
-  if (!missing(col)) colors <- col
+  if (!is.null(col)) colors <- col
+
+  if (is.null(scale_fill_gradient_fun)) {
+    if (node_type == "heatmap") {
```
Comment (talgalili) [inline R/heatmaply.R]: Both scale_fill_gradient_fun seem identical to me, so why the if (node_type == "heatmap")?

File: R/heatmapr.R (hunk 1 above)
Comment (talgalili) [inline R/heatmapr.R]: typo

File: R/zzz.R [hunk 6]
```diff
@@ -104,19 +104,19 @@ heatmaplyWelcomeMessage <- function() {
```
Comment (talgalili) [inline R/zzz.R]: These changes are rolling back some fixes I've made. Please change them back.

File: R/heatmaply.R (hunk 5 above)
Comment (Alanocallaghan) [reply R/heatmaply.R]: `scale_fill_gradientn` and `scale_color_gradientn`

**Comment by Alanocallaghan** [comment]
Sure - I still found a large diff after running stylr, most of it was unintentional. Would appreciate a heads up before reformatting large amounts of the code

File: R/heatmaply.R (hunk 5 above)
Comment (talgalili) [reply R/heatmaply.R]: now I see, thanks :)

**Comment by talgalili** [comment]
I'm sorry Alan, you're right, I should have emailed you at the time, I apologize.

Cheers,
Tal

File: R/zzz.R (hunk 6 above)
Comment (Alanocallaghan) [reply R/zzz.R]: My fault! Wasn't sure which was which :)

**Comment by Alanocallaghan** [comment]
My fault for leaving merge requests so long! Should be fixed now, but feel free to have another look. I'll merge at the weekend unless you have any objections

File: NEWS [hunk 7]
```diff
@@ -32,22 +46,16 @@ heatmaply 0.13.0 (2017-11-11)
...
-* heatmaply
-  - Add `cellnote_size` argument, controlling the font size of the cellnote.
-
 ### BUG FIXES
 * heatmaply
   - removing tick labels with `showticklabels` now removes the ticks as
     well as the tick labels.
   - Prevent grid_gap warning (#105)
+  - Add `cellnote_size` argument, controlling the font size of the cellnote.
```
Comment (talgalili) [inline NEWS]: I think this is a new feature and not a bug fix (please revert this edit to what was before in the file)

File: R/plots.R [hunk 8]
```diff
@@ -77,37 +85,99 @@ ggplot_heatmap <- function(xx,
   } else {
     assert_that(length(label_names) == 3)
   }
+  melt_df <- function(x, label_names) {
```
Comment (talgalili) [inline R/plots.R]: Might it be better to place this function outside of ggplot_heatmap?

**Comment by talgalili** [comment]
@Alanocallaghan 
Well, I'm not sure which of us is more responsible, but I'm glad we both think it is us and not the other one :)

I reviewed the code (although I didn't test that it works, but I trust you have), so I'm fine with you including it.
Also, if you could remove the lintr fork that would be better. In general, so to avoid future conflicts of code, I believe it would be better to try and keep only active forks (this way we could notify each other if we intend to do something big with the code).

Lastly, I believe the only major change I see in the future is deciding that the names of our function arguments are agreed upon (and changing the version to 1.0.0). I suspect that if we'd like to change anything, that would be a major code change that should not be done while having too active alternative forks.

Yours,
Tal

**Comment by talgalili** [comment]
Hi @Alanocallaghan - just wanted to ping and check if there is any issue with moving forward on merging this branch (it is not urgent by any means, I was just curious).

Yours,
Tal

**Comment by Alanocallaghan** [comment]
Hi tal,

Yep, will merge this week, apologies

On 17 Dec 2017 9:14 pm, Tal Galili <notifications@github.com> wrote:

Hi @Alanocallaghan<https://github.com/alanocallaghan> - just wanted to ping and check if there is any issue with moving forward on merging this branch (it is not urgent by any means, I was just curious).

Yours,
Tal

**Pull request merged**